Content.Shared/Explosion/ExplosionEvents.cs

```diff
@@ -23,38 +21,32 @@ public record struct GetExplosionResistanceEvent(string ExplosionPrototype) : II
 }
 
 /// <summary>
-/// Raised on an entity caught in an explosion to do damage to it any potentially add contents to also explode.
-/// Various systems add to contents such as inventory and storage.
-/// This is recursive so a matchbox in a backpack in a player's inventory will be handled.
+/// This event is raised directed at an entity that is about to receive damage from an explosion. It can be used to
+/// recursively add contained/child entities that should also receive damage. E.g., entities in a player's inventory
+/// or backpack. This event will be raised recursively so a matchbox in a backpack in a player's inventory
+/// will also receive this event.
 /// </summary>
 [ByRefEvent]
-public record struct ExplodedEvent
+public record struct BeforeExplodeEvent(DamageSpecifier Damage, string Id, List<EntityUid> Contents)
 {
-    public readonly DamageSpecifier? Damage;
-
-    public readonly float ThrowForce;
+    /// <summary>
+    /// The damage that will be received by this entity. Note that the entity's explosion resistance has already been
+    /// used to modify this damage.
+    /// </summary>
+    public readonly DamageSpecifier Damage = Damage;
 
     /// <summary>
     /// ID of the explosion prototype.
     /// </summary>
-    public readonly string Id;
+    public readonly string Id = Id;
 
     /// <summary>
-    /// Transform of the entity in the explosion the event was raised on.
+    /// Damage multiplier for modifying the damage that will get dealt to contained entities.
     /// </summary>
-    public readonly TransformComponent? Xform;
```

Comment:
I removed the xform argument because it wasn't being used and it avoids unnecessarily fetching the component

Reply:
its used by the player damage admin log, and is passed in by the caller of ProcessEntity anyway

Reply:
The player damage logging should eventually just use the "origin" entity uid argument in try change damage to log the damage & source, though that requires converting the explosion to an actual entity first.